frontend/src/metabase/nav/hooks.ts

```diff
@@ -0,0 +1,19 @@
+import { useDatabaseListQuery } from "metabase/common/hooks";
+import { useSelector } from "metabase/lib/redux";
+import { PLUGIN_SELECTORS } from "metabase/plugins";
+import { getIsPaidPlan } from "metabase/selectors/settings";
+import { getUserIsAdmin } from "metabase/selectors/user";
+
+export function useShouldShowDatabasePromptBanner() {
```

Comment:
It think it would be much easier (and more similar to how we do things in other places) to have a selector instead. Something like `getHasDatabasePromptBanner(state, { databases })`. It would be also much easier to test it.

Reply:
oh I was considering that too (sorry that I forgot to mention about that), but the problem is that there is a logic that uses to determine if we want to start fetching database list which is `isAdmin && isPaidPlan && !isWhiteLabeling` You can imagine that this won't be a problem if I just don't use `enabled` in `useDatabaseListQuery` at all, but I think it'd be a wast to have to call the API unnecessarily.

That from that point if I need to have this it means maybe I need 2 selectors
1. one that returns `isAdmin && isPaidPlan && !isWhiteLabeling` that I can use for the `enabled` prop
2. the one that actually will be used to determine whether or not we're gonna render the database prompt banner which is `isAdmin && isPaidPlan && !isWhiteLabeling && onlyHaveSampleDatabase`

It feels weird if I have to have only 1 selector `getHasDatabasePromptBanner(state, { databases })` but for the enabled props I still need to get `isAdmin`, `isPaidPlan`, and `isWhiteLabeling` individually. So I figure it's best to capture everything into a single hook.

Does this make sense?